src/main/kotlin/org/jetbrains/research/ml/codetracker/tracking/TaskFileHandler.kt

```diff
@@ -43,34 +44,27 @@ object TaskFileHandler {
         TaskDocumentListener()
     }
 
-    init {
-        if (PluginServer.serverConnectionResult != ServerConnectionResult.SUCCESS) {
-            subscribe(ServerConnectionNotifier.SERVER_CONNECTION_TOPIC, object : ServerConnectionNotifier {
-                override fun accept(connection: ServerConnectionResult) {
-                    // Todo: call it if the current language is chosen
-                    if (connection == ServerConnectionResult.SUCCESS) {
-                        projectsToInit.forEach { initProject(it) }
-                        projectsToInit.clear()
-                    }
-                }
-            })
-        }
+    fun initProjects() {
+        projectsToInit.forEach { initProject(it) }
     }
 
     /**
-     * Call if you sure that ServerConnectionResult was successful and therefore all tasks are received
+     * Call each time when user change the language
```

Comment:
Are you sure about this comment? It seems you call each time the user press `startWorkingButton`

Reply:
Of course, you are right! Thank you